jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/kafka/listeners/Asn1DecodedDataRouter.java

```diff
@@ -0,0 +1,194 @@
+package us.dot.its.jpo.ode.kafka.listeners;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.json.JSONObject;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+import us.dot.its.jpo.ode.coder.OdeBsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeMapDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdePsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSpatDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSrmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeSsmDataCreatorHelper;
+import us.dot.its.jpo.ode.coder.OdeTimDataCreatorHelper;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.kafka.producer.DisabledTopicException;
+import us.dot.its.jpo.ode.kafka.topics.JsonTopics;
+import us.dot.its.jpo.ode.kafka.topics.PojoTopics;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeBsmData;
+import us.dot.its.jpo.ode.model.OdeLogMetadata;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.plugin.j2735.J2735DSRCmsgID;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+/**
+ * The Asn1DecodedDataRouter class is a component responsible for processing decoded ASN.1 data
+ * from Kafka topics. It listens to messages on a specified Kafka topic and handles the incoming
+ * data by processing and forwarding it to different topics based on specific criteria.
+ *
+ * <p>This listener is specifically designed to handle decoded data produced by the asn1_codec.
+ * Upon receiving a payload, it uses transforms the payload and then determines the appropriate
+ * Kafka topic to forward the processed data.</p>
+ *
+ * <p>The class utilizes Spring Kafka's annotation-driven listener configuration,
+ * allowing it to automatically consume messages from a configured Kafka topic.</p>
+ */
+@Slf4j
+@Component
+public class Asn1DecodedDataRouter {
+
+  private final PojoTopics pojoTopics;
+  private final JsonTopics jsonTopics;
+  private final KafkaTemplate<String, String> kafkaTemplate;
+  private final KafkaTemplate<String, OdeBsmData> bsmDataKafkaTemplate;
+
+  /**
+   * Constructs an instance of Asn1DecodedDataRouter.
+   *
+   * @param kafkaTemplate the KafkaTemplate used for sending messages to Kafka topics.
+   */
+  public Asn1DecodedDataRouter(KafkaTemplate<String, String> kafkaTemplate,
```

Comment:
(non-blocking) Is the `@Autowired` annotation necessary here, or does the class `@Component` annotation allow Spring to wire everything up without it?

Reply:
The `@Component` annotation indicates to Spring the `Asn1DecodedDataRouter` should be instantiated by Spring, and that other Spring beans provide the parameters to the constructor. That's the long way to say, yea, we don't need the `@Autowired` annotation here

Reply:
Thank you for the explanation!